src/User/Controller/SettingsController.php

```diff
@@ -488,4 +547,81 @@ protected function disconnectSocialNetwork($id)
         $account->delete();
         $this->trigger(SocialNetworkConnectEvent::EVENT_AFTER_DISCONNECT, $event);
     }
+
+    /**
+     * This action is called when the user is forced to use two factor authentication based on his role.
+     * The action calls the service that enables two factor authentication for the user.
+     * In more details, it opens a modal window that shows the QRcode that have to be scanned by Google Authenticator o Authy.
+     * The app generates a code that the user has to insert in the form to enable two factor authentication.
+     * The user is enabled to two factor authentication by the action actionTwoFactorEnableForced.
+     *
+     * @param type $id
+     * 
+     * @return type
+     */
+    public function actionTwoFactorForced($id)
```

Comment:
Are this and the following action still required, since the implementation of the filter?